### PR TITLE
Sphinx gh pages

### PR DESCRIPTION
**Pull request by hndgzkn**
Deploy documentation generated with Sphinx to github pages.

Contains Examples and API sections. 

API documentation is to be updated with relevant modules/functions.

### CODE REVIEW

**Comment by agramfort** [comment]
why are CIs not running? it would be great to see the rendered doc as an artifact from the CI

**Comment by rprimet** [comment]
@agramfort I see jobs being run in https://github.com/hndgzkn/dicodile/actions ?

**Comment by agramfort** [comment]
but not in the PR. maybe cause the PR is in WIP stage?

**Comment by hndgzkn** [comment]
Shall I put `on: pull_request`  in the actions file? 

As this job creates and pushes artifacts to gh_pages branch in the repo, I couldn't be sure about the behaviour.

**Comment by agramfort** [comment]
the artifact should be built on the PR but the push to gh-pages should be
done only from master branch

>

**Review by agramfort — APPROVED** [review]
I can see the artifact. Works great. Let's merge this and iterate

thanks @hndgzkn

File: README.md [hunk 1]
```diff
@@ -1,25 +0,0 @@
-![Build Status](https://github.com/tomMoral/dicodile/workflows/unittests/badge.svg)
-[![codecov](https://codecov.io/gh/tomMoral/dicodile/branch/master/graph/badge.svg)](https://codecov.io/gh/tomMoral/dicodile)
```
Comment (tomMoral) [inline README.md:2]: We need to add this in the `README.rst`.

File: docs/api.rst [hunk 2]
```diff
@@ -0,0 +1,13 @@
+.. _api_documentation:
+
+=================
+API Documentation
+=================
+
+.. currentmodule:: dicodile
```
Comment (tomMoral) [inline docs/api.rst:7]: For the API, I think that for now we should only expose the description for the `dicodile` function inside this module, as it is the main one that will be used.

**Review by tomMoral — COMMENTED** [review]
A couple of comments.

File: docs/requirements.txt [hunk 3]
```diff
@@ -0,0 +1,3 @@
+numpydoc
+sphinx_bootstrap_theme
+sphinx_gallery
```
[the remark this answers is not in the document]
Comment (hndgzkn) [reply docs/requirements.txt:3]: Sounds good to me. I can put dependencies in setup.config and build the docs without using the action.

[the remark this answers is not in the document]
Comment (tomMoral) [reply docs/requirements.txt:3]: I think there is a `pre_build` parameter in the action wo you can probably keep the action but simply use this parameter to install using `pip install -e .[doc]`

[the remark this answers is not in the document]
Comment (hndgzkn) [reply docs/requirements.txt:3]: Ah yes, thanks!

**Comment by tomMoral** [comment]
Ok look good to me know but not sure why it failed!

**Comment by tomMoral** [comment]
Thx a lot @hndgzkn ! :)

**Pull request merged**